lib/experimental/fonts-library/class-wp-font-family.php

```diff
@@ -0,0 +1,421 @@
+			: array();
+		$this->data     = $sanitized_font;
+		return $this->data;
+	}
+
+	/**
+	 * Downloads font face assets.
+	 *
+	 * Downloads the font face asset(s) associated with a font face. It works with both single
+	 * source URLs and arrays of multiple source URLs.
+	 *
+	 * @param array $font_face The font face array containing the 'src' attribute with the source URL(s) of the assets.
+	 * @return array The modified font face array with the new source URL(s) to the downloaded assets.
+	 */
+	private function download_font_face_assets( $font_face ) {
+		$new_font_face        = $font_face;
+		$srcs                 = ! empty( $font_face['src'] ) && is_array( $font_face['src'] )
+			? $font_face['src']
+			: array( $font_face['src'] );
+		$new_font_face['src'] = array();
+		$i                    = 0;
+		foreach ( $srcs as $src ) {
+			$filename = WP_Font_Family_Utils::get_filename_from_font_face( $font_face, $src, $i++ );
+			$new_src  = $this->download_asset( $src, $filename );
+			if ( $new_src ) {
+				$new_font_face['src'][] = $new_src;
+			}
+		}
+		if ( count( $new_font_face['src'] ) === 1 ) {
+			$new_font_face['src'] = $new_font_face['src'][0];
+		}
+		return $new_font_face;
+	}
+
+
+	/**
+	 * Downloads font face assets if the font family is a Google font, or moves them if it is a local font.
+	 *
+	 * @param array $files An array of files to be installed.
+	 * @return bool|WP_Error
+	 */
+	private function download_or_move_font_faces( $files ) {
+		if ( $this->has_font_faces() ) {
+			$new_font_faces = array();
+			foreach ( $this->data['fontFace'] as $font_face ) {
+				if ( empty( $files ) ) {
+					// If we are installing local fonts, we need to move the font face assets from the temp folder to the wp fonts directory.
+					$new_font_face = $this->download_font_face_assets( $font_face );
+				} else {
+					// If we are installing google fonts, we need to download the font face assets.
+					$new_font_face = $this->move_font_face_asset( $font_face, $files[ $font_face['file'] ] );
+				}
+				// If the font face assets were successfully downloaded, we add the font face to the new font.
+				// Font faces with failed downloads are not added to the new font.
+				if ( ! empty( $new_font_face['src'] ) ) {
+					$new_font_faces[] = $new_font_face;
+				}
+			}
+			if ( ! empty( $new_font_faces ) ) {
+				$this->data['fontFace'] = $new_font_faces;
+				return true;
+			}
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Get the post for a font family.
+	 *
+	 * @return WP_Post|null The post for this font family object or null if the post does not exist.
+	 */
+	public function get_font_post() {
+		$args = array(
+			'post_type'      => 'wp_fonts_library',
+			'post_name'      => $this->data['slug'],
+			'name'           => $this->data['slug'],
+			'posts_per_page' => 1,
+		);
+
+		$posts_query = new WP_Query( $args );
+
+		if ( $posts_query->have_posts() ) {
+			$post = $posts_query->posts[0];
+			return $post;
```

Comment:
```suggestion
			return $posts_query->posts[0];
```
No need to store this in a variable before returning.